kb(drawer): refresh a chart inside a drawer (#2061)
telerik/blazor-docs: knowledge-base/drawer-chart-refresh.md

---
title: Refresh a Chart Inside a Drawer
description: How to refresh a Chart within a Drawer content.
type: how-to
page_title: Refresh a Chart Inside a Drawer
slug: drawer-kb-chart-refresh
position: 
tags: 
ticketid: 1583102
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Drawer for Blazor,<br /> Charts for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

This knowledge base article answers the following questions:

* How do I refresh a Chart inside a Drawer?
* Is it possible to refresh a Chart when the Drawer is expanded?

## Solution

> 1. Create a bool flag and set it within the Drawer's [`ExpandedChanged`]({%slug drawer-events%}#expandedchanged) handler to indicate when the Chart needs to be refreshed

Reviewer: Both list items (1. and 2.) need ending punctuation as per [our style guide](https://docs.telerik.com/style-guide/punctuation#lists).

As per our style guide, active voice is preferable over passive voice, especially if you're mixing the two within your sentence. Consider going with `to indicate when the Chart requires a refresh`.

> 2. Call the Chart's [`Refresh()` method]({%slug components/chart/overview%}#chart-reference-and-methods) within the `OnAfterRenderAsync` lifecycle method after a brief delay that allows the Drawer to finish expanding

Reviewer: Personally, I would split the sentence in two for conciseness and clarity. For example:

```suggestion
2. Call the Chart's [`Refresh()` method]({%slug components/chart/overview%}#chart-reference-and-methods) within the `OnAfterRenderAsync` lifecycle method. Set a brief delay to allow the Drawer to finish expanding before the refresh.
```
